libdebug/libdebug.py

```diff
@@ -917,33 +943,54 @@ def syscalls_to_not_pprint(self: _InternalDebugger, value: list[int] | list[str]
             self._enable_pretty_print()
 
     @property
-    def signal_to_pass(self: _InternalDebugger) -> list[str]:
-        """Get the signal to pass to the process.
+    def signal_to_block(self: _InternalDebugger) -> list[str]:
+        """Get the signal to not forward to the process.
 
         Returns:
-            list[str]: The signals to pass.
+            list[str]: The signals to block.
         """
-        return [resolve_signal_name(v) for v in self.context._signal_to_pass]
+        return [resolve_signal_name(v) for v in self.context._signal_to_block]
 
-    @signal_to_pass.setter
-    def signal_to_pass(self: _InternalDebugger, signals: list[int] | list[str]) -> None:
-        """Set the signal to pass to the process.
+    @signal_to_block.setter
+    def signal_to_block(
+        self: _InternalDebugger, signals: list[int] | list[str]
+    ) -> None:
+        """Set the signal to not forward to the process.
 
         Args:
-            signals (list[int] | list[str]): The signals to pass.
+            signals (list[int] | list[str]): The signals to block.
         """
         if not isinstance(signals, list):
-            raise TypeError("signal_to_pass must be a list of integers or strings")
+            raise TypeError("signal_to_block must be a list of integers or strings")
 
-        signals = [v if isinstance(v, int) else resolve_signal_number(v) for v in signals]
+        signals = [
+            v if isinstance(v, int) else resolve_signal_number(v) for v in signals
+        ]
 
         if not set(signals).issubset(get_all_signal_numbers()):
             raise ValueError("Invalid signal number.")
 
-        self.context._signal_to_pass = signals
+        self.context._signal_to_block = signals
+
+    @property
+    def threads(self: _InternalDebugger) -> ThreadList:
```

Comment:
Do we really need properties for these attributes?

Reply:
What would you do? It should be only in read mode

Reply:
It's Python, nothing is really read-only

Reply:
I use properties whenever I need to validate the value passed to the setter, in other cases regular attributes serve the same purpose.
I don't see the need for that here and in some of the following properties

Reply:
I do not see your point. The behavior of having an @attribute without the setter is significantly different.

If you have a property of that type and try something like d.thread_id = 0, you would encounter the following clear and useful error for the user:
![image](https://github.com/libdebug/libdebug/assets/58036457/918b4084-2cf6-42fc-b2a3-6ea4abf3b190)

So yes, it is like having a setter that do not allow you to set the attribute. 

Without property, no errors are raised and python set the attribute. It is confusing and close to be impossible to debug for the user.

Reply:
I do not see why we need to enforce validation for these fields.
The user would never call
```
d.thread_id = 0
```
in any reasonable scenario. Do we really need to enforce that this must not be called?

Reply:
There's nothing that would make the user think that this attribute is editable

Reply:
If you want to go down the route of attribute validation for everything, we have a billion public attributes we expose that could break everything if inadvertently changed by the user

Reply:
Indeed, I am gradually enforcing the sensitive parameters as read-only. I'm not concerned about all parameters, but those that could break the debugger due to errors that are very difficult to understand. It is a decorator, and I really do not understand why this bothers you. It’s a very slight code change with almost null cost but could make the difference in avoiding hours of debugging a broken script.

Reply:
I mean, I do not see the disadvantages that are not pure software engineering wars.